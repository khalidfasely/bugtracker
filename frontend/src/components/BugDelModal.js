import Modal from "react-modal";

const BugDelModal = ({ delModalOpen, setDelModalOpen, deleteBug }) => (
    <Modal
        isOpen={delModalOpen}
        onRequestClose={() => setDelModalOpen(false)}
    >
        If you delete this bug there is no way to return it!
        <button onClick={deleteBug} data-testid='delete_button'>Delete</button>
    </Modal>
);

export default BugDelModal;