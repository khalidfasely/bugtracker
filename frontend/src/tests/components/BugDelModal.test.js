import React from "react";
import { render, fireEvent } from '@testing-library/react';

import BugDelModal from '../../components/BugDelModal';

let props;

beforeEach(() => {
    props = {
        delModalOpen: true,
        setDelModalOpen: jest.fn(),
        deleteBug: jest.fn()
    };
});

test('Should render BugDelModal component correctly', () => {
    const { asFragment } = render(<BugDelModal {...props} />);
    expect(asFragment()).toMatchSnapshot();
});

test('Should render/fire delete button correctly', () => {
    const { getByTestId } = render(<BugDelModal {...props} />);
    const deleteButtonEl = getByTestId('delete_button');
    expect(deleteButtonEl.textContent).toBe('Delete');

    expect(props.deleteBug).toHaveBeenCalledTimes(0);

    fireEvent.click(deleteButtonEl);

    expect(props.deleteBug).toHaveBeenCalledTimes(1);
});