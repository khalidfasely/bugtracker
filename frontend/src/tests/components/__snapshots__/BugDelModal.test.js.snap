// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render BugDelModal component correctly 1`] = `
<DocumentFragment>
  If you delete this bug there is no way to return it!
  <button
    data-testid="delete_button"
  >
    Delete
  </button>
</DocumentFragment>
`;
